test/unnest.js

```diff
@@ -1,28 +1,13 @@
-var assert = require('assert');
+var S = require('sanctuary');
 
 var R = require('..');
 var eq = require('./shared/eq');
-var Maybe = require('./shared/Maybe');
 
 
 describe('unnest', function() {
 
   it('only flattens one layer deep of a nested list', function() {
-    var nest = [1, [2], [3, [4, 5], 6, [[[7], 8]]], 9, 10];
-    eq(R.unnest(nest), [1, 2, 3, [4, 5], 6, [[[7], 8]], 9, 10]);
-    nest = [[[[3]], 2, 1], 0, [[-1, -2], -3]];
-    eq(R.unnest(nest), [[[3]], 2, 1, 0, [-1, -2], -3]);
-    eq(R.unnest([1, 2, 3, 4, 5]), [1, 2, 3, 4, 5]);
```

Comment:
I agree that the rest of these can go.  But I think this edge-case provides a useful description of the operation.  What happens if there are *no* sublists?

Reply:
> What happens if there are *no* sublists?

Let's ask Haskell:

```
> import Control.Monad

> :t join
join :: Monad m => m (m a) -> m a

> join [[[1], [2]], [[3], [4]]]
[[1], [2], [3], [4]]

> join [[1, 2], [3, 4]]
[1, 2, 3, 4]

> join [1, 2, 3, 4]
    • Non type-variable argument in the constraint: Num [a]
      (Use FlexibleContexts to permit this)
    • When checking the inferred type
        it :: forall a. Num [a] => [a]
```

Let's ask Sanctuary:

```
> S.join
join :: Chain m => m (m a) -> m a

> S.join([[[1], [2]], [[3], [4]]])
[[1], [2], [3], [4]]

> S.join([[1, 2], [3, 4]])
[1, 2, 3, 4]

> S.join([1, 2, 3, 4])
TypeError: Invalid value

join :: Chain m => m (m a) -> m a
                     ^^^^^
                       1

1)  1 :: Number, FiniteNumber, NonZeroFiniteNumber, Integer, ValidNumber

The value at position 1 is not a member of ‘m a’.
```

My understanding is that we do not specify the behaviour of a function when applied to invalid arguments. In this case, `[1, 2, 3, 4, 5]` is an invalid argument.

Reply:
I think I can buy that.  I may give it a little more thought, though.